services/121-service/src/registration/registrations.service.ts

```diff
@@ -389,28 +351,26 @@ export class RegistrationsService {
     );
   }
 
-  public async importRegistrationFromJson(
-    validatedJsonData: ImportRegistrationsDto[],
+  public async importRegistrationsFromJson(
+    jsonData: Record<string, string | number | boolean>[],
     programId: number,
     userId: number,
   ): Promise<ImportResult> {
     const program = await this.findProgramOrThrow(programId);
-    if (!program?.published) {
+    this.throwIfProgramIsNotPublished(program.published);
+    return await this.registrationsImportService.importRegistrations(
```

Comment:
Do we not want to check first for the correct contents of the json before calling this? As the old code did check (?) based on the Dto?

Reply:
not completely sure what you mean. The validation of JSON of the new registrations need the programId to be able to validate the required program registration attributes of a program. So better to first validate if the program stuff is correct, also these program checks are less expensive